suggester/src/main/java/org/opengrok/suggest/Suggester.java

```diff
@@ -278,8 +279,12 @@ public void rebuild(final Collection<NamedIndexDir> indexDirs) {
         rebuildLock.unlock();
 
         synchronized (lock) {
+            if (terminating) {
```

Comment:
What would you think about also short-circuiting the init- and rebuild-runnables to check `terminating`?

Reply:
Inside the runnables the benefit is small I think however checking it in `init()` makes sense.

Reply:
The runnables are the items potentially in queue. It makes sense for them to be `terminating`-aware so that when they're awakened they do no unnecessary work. If you have a for-loop that is just submitting to an asynchronous executor (as this code does), you have a minuscule amount of time where being `terminating`-aware has a chance of being effective.

Moreover, `shutdown()` for a work-stealing pool only prevent new tasks from being submitted; it is an "orderly shutdown " that allows all submitted tasks to be executed. So again, the runnables should be `terminating`-aware for there to be any realistic chance of avoiding work.

I recommend reverting to your commit 62b8e45 and just addressing the runnables.

Reply:
You're right, addressing this in the runnables is better and actually more readable.